ansible/roles/ci/code_style_check/common_resources/files/files_by_extension.py

```diff
@@ -37,29 +37,78 @@ def argparser():
 
 
 def gather_all_python_files(file_path):
+    """This function walks through the given path collects all python files (with and without extension based on
+       header). It will exclude all python files found in the lint ignore file."""
     output = []
     for dirpath, _, filenames in os.walk(file_path):
+        excluded_files = gather_excluded_files(dirpath)  # Gathers all the files to be ignored in this folder
         for filename in filenames:
-            file_path = f"{dirpath}/{filename}"
-            try:
-                fline = ""
-                with open(file_path) as python_file:
-                    fline = python_file.readline().strip()
-                if any(fline in head for head in PYTHON_HEADERS) and fline != "" and fline != "#":
-                    output.append(file_path)
-            except Exception:  # We don't actually care about the exception as it wont fail on python files.
+            if filename in excluded_files:  # Skip excluded files
                 continue
+            full_file_path = os.path.join(dirpath, filename)
+            valid_file = check_is_python_file(full_file_path)
+            if valid_file:
+                output.append(full_file_path)
     return output
 
 
 def gather_all_cpp_files(file_path):
+    """This function walks through the given path collects all c, cpp, h, hpp files It will exclude all python files
+       found in the lint ignore file."""
     output = []
     for dirpath, _, filenames in os.walk(file_path):
+        excluded_files = gather_excluded_files(dirpath)  # Gathers all the files to be ignored in this folder
         for filename in filenames:
+            if filename in excluded_files:  # Skip excluded files
+                continue
             if os.path.splitext(filename)[1] in CPP_HEADERS:
-                output.append(f"{dirpath}/{filename}")
+                output.append(os.path.join(dirpath, filename))
     return output
 
+def gather_excluded_files(folder_path):
+    """Gatheres all of the files to ignore in the lint ignore file (if it exists). If the file contains
+       `exclude_files=*` then all files will be ignored, or you can list indiviual files like
+       `exclude_files=test.py,hello.cpp,hi.h`."""
+    folder_files = [folder_file for folder_file in os.listdir(folder_path)
+                    if os.path.isfile(os.path.join(folder_path, folder_file))]
+    folder_files_filtered = []
```

Comment:
You could probably move these lines also to the if LINT_IGNORE_FILE block

Reply:
I can't as it's checking if the file exists in folder_files